Microsoft.Toolkit.Uwp.UI/Media/RadialGradientBrush.cs

```diff
@@ -103,5 +105,37 @@ protected override bool OnDraw(CanvasDevice device, CanvasDrawingSession session
 
             return false;
         }
+
+        private void RegisterAppResumeHandler()
+        {
+            var device = CanvasDevice.GetSharedDevice();
+            device.DeviceLost += RadialGradientBrush_DeviceLost;
+            CanvasComposition.CreateCompositionGraphicsDevice(Window.Current.Compositor, device).RenderingDeviceReplaced += RadialGradientBrush_RenderingDeviceReplaced;
+        }
+
+        private void RadialGradientBrush_RenderingDeviceReplaced(Windows.UI.Composition.CompositionGraphicsDevice sender, Windows.UI.Composition.RenderingDeviceReplacedEventArgs args)
+        {
+            sender.RenderingDeviceReplaced -= RadialGradientBrush_RenderingDeviceReplaced;
+            OnConnected();
+            OnDisconnected();
+        }
+
+        private void RadialGradientBrush_DeviceLost(CanvasDevice sender, object args)
+        {
+            sender.DeviceLost -= RadialGradientBrush_DeviceLost;
+            OnConnected();
+            OnDisconnected();
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the <see cref="RadialGradientBrush"/> class.
+        /// Remove the event handlers when the app is closed.
+        /// </summary>
+        ~RadialGradientBrush()
+        {
+            var device = CanvasDevice.GetSharedDevice();
+            device.DeviceLost -= RadialGradientBrush_DeviceLost;
+            CanvasComposition.CreateCompositionGraphicsDevice(Window.Current.Compositor, device).RenderingDeviceReplaced -= RadialGradientBrush_RenderingDeviceReplaced;
```

Comment:
You are allocating an object on the destructor. Store it's value on a variable and clear everything on the same object. By calling CreateCompositionGraphicsDevice you are creating another object. This will not free the other instance handler.

Reply:
I need to create a new method to do this and call it from the destructor?